src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/TargetCompatibilityReferenceAnalyzer.cs

```diff
@@ -65,16 +68,23 @@ public async Task AnalyzeAsync(IProject project, IDependencyAnalysisState state,
 
                         if (isMajorChange)
                         {
+                            details.Add(string.Concat("Package ", packageReference.Name, " needs to be upgraded across major versions (", packageReference.Version, " -> ", updatedReference.Version, ") which may introduce breaking changes"));
                             _logger.LogWarning("Package {NuGetPackage} has been upgraded across major versions ({OldVersion} -> {NewVersion}) which may introduce breaking changes", packageReference.Name, packageReference.Version, updatedReference.Version);
                         }
 
                         if (updatedReference.IsPrerelease)
                         {
+                            details.Add(string.Concat("Package ", packageReference.Name, " needs to be upgraded to a prerelease version (", updatedReference.Version, ") because no released version supports target(s) ", string.Join(", ", state.TargetFrameworks)));
```

Comment:
This is a bunch of string concatenation going on here. Can you swap it out to use string.Format (or better yet, SR.Format)? Maybe also consider using resource files